File: app/src/main/java/com/example/domain/type/FizzBuzzType03.java
package com.example.domain.type;

import com.example.domain.model.FizzBuzzValue;

public class FizzBuzzType03 extends FizzBuzzType {
  public FizzBuzzValue generate(int number) {
    if (fizz(number) && buzz(number)) {
      return new FizzBuzzValue(number, "FizzBuzz");
    }
    return new FizzBuzzValue(number, String.valueOf(number));
  }
}

File: app/fizzbuzz-scala/src/main/scala/domain/types/FizzBuzzType03.scala
package domain.types

import domain.model.FizzBuzzValue

object FizzBuzzType03 extends FizzBuzzType {
  override def generate(number: Int): FizzBuzzValue = {
    if (isFizzBuzz(number)) {
      FizzBuzzValue(number, "FizzBuzz")
    } else {
      FizzBuzzValue(number, number.toString)
    }
  }
}
